File: javascript-exercises-master/calculator/calculator.js
function add (a, b) {
    return a + b;
	
}

function subtract (a, b) {
	return a - b;
}

function sum (arr) {
	let sum = 0;
	let item;
	for(item of arr){
	    if(!isNaN(item)){
	        sum += item;
        }
    }
    return sum;
}

function multiply (arr) {
    let prod = 1;
    let item;
    for(item of arr){
        if(!isNaN(item)){
            prod *= item;
        }
    }
    return prod;
}

function power(x, y) {
	return Math.pow(x, y);
}

function factorial(x) {
	if(x === 0) return 1;
	if(x === 1) return 1;
	return(x * factorial(x-1));
}

module.exports = {
	add,
	subtract,
	sum,
	multiply,
    power,
	factorial
}

File: learn-ruby/02_calculator/calculator.rb
def add(x,y)
  x+y
end

def subtract(x,y)
  x-y
end

def sum(arr)
  unless arr.empty?
    return arr.reduce(:+)
  end
  0
end

def multiply(*args)
  if args.size == 1
    return args[0].reduce(:*)
  end
  args.reduce(:*)
end

def power(x, y)
  x**y
end

def factorial(n)
  return 1 if n == 0
  multiply 1..n
end